tensorflow/lite/tools/reverse_xxd_dump_from_cc.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+r"""This tool reverses xxd dump from *.cc source file to its original binary file
+
+This script is used to convert models from C++ source file (dumped with xxd) to
+the binary model weight file and analyze it with model visualizer like Netron
+(https://github.com/lutzroeder/netron) or load the model in TensorFlow Python API
+to evaluate the results in Python.
+
+The command to dump binary file to C++ source file looks like
+
+xxd -i model_data.tflite > model_data.cc
+
+Example usage:
+
+python reverse_xxd_dump_from_cc.py model_data.cc --output=model_data.tflite
+"""
+import argparse
+import os
+import re
+
+
+def generate_default_output(filename, postfix=None, extension=None):
+  """Generate output filename given the filename and extension
+
+  Args:
+    filename(str): Input filename
+    postfix(str): Postfix to add to the output filename
+    extension(str): Output file extension, if not given, it will be
+      the same as input file.
+
+  Return:
+    string for the output filename given input args
+  """
+  name, ext = os.path.splitext(filename)
+
+  if extension is not None:
+    if not extension.startswith("."):
+      extension = "." + extension
```

Comment:
the sematics is confusing and it isn't used right now. I'd say drop the extension parameter (actually the entire `generate_default_output function`).

Reply:
Gotcha, done!